### PR TITLE
Add conversion to PEM

### PR DESCRIPTION
**Pull request by pathetic-lynx**
Detect when the key files are not in PEM format and convert them if necessary

### CODE REVIEW

**Comment by pathetic-lynx** [comment]
I can confirm this works for rsa, ecdsa and ed25519.
The detection with grep seems a little sketchy to me, not sure if this always correct.

File: modules/60crypt-ssh/module-setup.sh [hunk 1]
```diff
@@ -78,7 +78,13 @@ install() {
         cp ${state}.pub ${osshKey}.pub
         ;;
     esac
-    
+
+    grep -q -- "-----BEGIN ${msgKeyType} PRIVATE KEY-----" ${osshKey} || {
+      ssh-keygen -q -e -p -P "" -N "" -m pem -f ${osshKey} &> /dev/null
+          && dinfo "Converting ${osshKey} to PEM format"
+          || derror "Converting ${osshKey} to PEM format failed"
+    }
```
Comment (ahesford) [inline modules/60crypt-ssh/module-setup.sh:86]: Eventually, the entire key-handling portion should be reworked. OpenSSH keys should be a fallback, not the fundamental currency of this module. The SYSTEM option is a bad idea and should be de-emphasized or removed entirely. Generated keys should persist between runs (or at least a configuration option should allow this).

In the meantime, the multiple nesting of Boolean operators is a bit hard to parse, and let's make the header parsing a bit stricter.

```suggestion
    if ! head -n1 "${osshKey}" | grep -q -- "-----BEGIN ${msgKeyType} PRIVATE KEY-----"; then
      dinfo "Converting ${osshKey} to PEM format"
      ssh-keygen -q -e -p -P "" -N "" -m pem -f ${osshKey} > /dev/null 2>&1 || \
        derror "Converting ${osshKey} to PEM format failed"
    fi
```

**Review by ahesford — CHANGES_REQUESTED** [review]

**Comment by ahesford** [comment]
Closing due to inactivity
